src/main/java/twilightforest/block/BlockTFMagicLogSpecial.java

```diff
@@ -186,40 +188,33 @@ private void doMinersTreeEffect(World world, BlockPos pos, Random rand) {
 	private void doSortingTreeEffect(World world, BlockPos pos, Random rand) {
 
 		// find all the chests nearby
-		List<IInventory> chests = new ArrayList<>();
+		List<IItemHandlerModifiable> inventories = new ArrayList<>();
 		int itemCount = 0;
 
 		for (BlockPos iterPos : WorldUtil.getAllAround(pos, 16)) {
+			IItemHandlerModifiable handler = null;
 
-			IInventory chestInventory = null, teInventory = null;
+			TileEntity te = world.getTileEntity(iterPos);
+			if (te == null) continue;
 
-			Block block = world.getBlockState(iterPos).getBlock();
-			if (block instanceof BlockChest) {
-				chestInventory = ((BlockChest) block).getContainer(world, iterPos, true);
-			}
+			if (!te.hasCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, EnumFacing.UP)) continue;
 
-			TileEntity te = world.getTileEntity(iterPos);
-			if (te instanceof IInventory && !te.isInvalid()) {
-				teInventory = (IInventory) te;
-			}
+			handler = (IItemHandlerModifiable) te.getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, EnumFacing.UP);
```

Comment:
not every IItemHandler is IItemHandlerModifiable, so we can't assume IItemHandlerModifiable.

In general, you'll have to change the implementation from writing specific slots to just using the insert/extract helpers in `ItemHandlerHelper`

Reply:
Thanks! I didn't think it would be this simple, but I wasn't familiar enough with the code.

Is it safe to assume that all chest and chest equivalent and chest-adjacent chests would be IItemHandlerModifiable? Ergo, testing the interface to see if it is modifiable and skipping it if it isn't might be sufficient?